src/components/Filters/SchemaSieve.ts

```diff
@@ -1,59 +1,105 @@
+import { OperatorSlug as dateTimeOperatorSlug } from '../DataTypes/date-time';
+import { OperatorSlug as enumOperatorSlug } from '../DataTypes/enum';
+import { OperatorSlug as numberOperatorSlug } from '../DataTypes/number';
+import { OperatorSlug as objectOperatorSlug } from '../DataTypes/object';
+import { OperatorSlug as oneOfOperatorSlug } from '../DataTypes/oneOf';
+import { OperatorSlug as stringOperatorSlug } from '../DataTypes/string';
 
 const ajv = new Ajv();
 ajvKeywords(ajv, ['regexp', 'formatMaximum', 'formatMinimum']);
 
 export const FULL_TEXT_SLUG = 'full_text_search';
-const DEFAULT_DELIMITER = '___';
+export interface Operator<T> {
+	slug: T;
+	label: string;
+}
+
+export type OperatorSlugs =
+	| arrayOperatorSlug
+	| booleanOperatorSlug
+	| dateTimeOperatorSlug
+	| enumOperatorSlug
+	| numberOperatorSlug
+	| objectOperatorSlug
+	| oneOfOperatorSlug
+	| stringOperatorSlug;
+
+export type CreateFilter = (
+	field: string,
+	operator: Operator<OperatorSlugs>,
+	value: any,
+	subSchema: JSONSchema,
+	recursive: boolean,
+	refScheme?: string,
+) => JSONSchema;
+
+export type EditSchema = (
+	schema: JSONSchema,
+	operator: Operator<OperatorSlugs> | undefined,
+) => JSONSchema;
+
 interface FullTextFilterItem {
 	key: string;
 	schema: JSONSchema;
-	operator: string;
+	operator: Operator<string>;
 	model: any;
 	searchKey?: string;
 }
 
-export function filter<T>(
-	filters: JSONSchema | JSONSchema[],
-	collection: T[],
-): T[];
-export function filter<T>(
-	filters: JSONSchema | JSONSchema[],
-	collection: Record<string, T>,
-): Record<string, T>;
-export function filter<T>(
-	filters: JSONSchema | JSONSchema[],
-	collection: T[] | Record<string, T>,
-) {
-	// Remove all schemas that may have been compiled already
-	ajv.removeSchema(/^.*$/);
-
-	const validators = Array.isArray(filters)
-		? filters.map((s) => ajv.compile(s))
-		: [ajv.compile(filters)];
+export const getSignatures = (filter: JSONSchema): FilterSignature[] => {
+	const signatures = filter.anyOf!.map((f: JSONSchema) => {
+		return parseFilterDescription(f);
+	});
 
-	if (Array.isArray(collection)) {
-		return collection.filter((m) => every(validators, (v) => v(m)));
-	}
+	return (signatures.filter((s) => s !== null) ?? []).map((s) => ({
+		...s,
+		key: !!s.refScheme ? `${s.field}__${s.refScheme}` : s.field,
+	})) as FilterSignature[];
+};
 
-	return pickBy(collection, (m) => every(validators, (v) => v(m)));
-}
+export const createFilter = (
+	schema: JSONSchema,
+	signatures: FilterSignature[],
+): JSONSchema => {
+	const anyOf = signatures.map(({ field, operator, value, refScheme }) => {
+		const subSchema = !!refScheme
+			? generateSchemaFromRefScheme(field, schema, refScheme)
+			: { ...(schema.properties![field] as JSONSchema) };
+		const model = getDataModel(subSchema);
+		if (!model) {
+			return {};
+		}
+		return (model.createFilter as CreateFilter)(
+			field,
+			operator,
+			value,
+			subSchema,
+			false,
+			refScheme,
+		);
```

Comment:
Should we v ?
* Remove the `recursive` argument from `getBase`
* Stop calling getBase from within the DataTypes, in favor of calling it only from here.

```suggestion
		return {
			...getBase(
				// recursive, // drop me
				operator,
				key,
				title,
				field,
				value,
				refScheme,
			),
			...(model.createFilter as CreateFilter)(
				field,
				operator,
				value,
				subSchema,
				false,
				refScheme,
			),
		};
```
And assuming that this will be the only place that `getBase` would be called, we could even inline it here.